### PR TITLE
Update to Compose 1.4

### LINKED ISSUE #261: Updates for Compose 1.4
- [x] TextField minLines - https://github.com/androidx/androidx/commit/e74bce3b62ac62a8115f7707bf5b58adc8d2c340
- [x] use IME animationTarget to properly delay TextField shift (alternatively use latest auto-bring-into-view behavior)
- [x] progress bar stroke cap -> https://github.com/androidx/androidx/commit/047b8e3359131d1317ecded79ad5028c009908b4
- [x] isContainer on semantics -> https://github.com/androidx/androidx/commit/5361cd861c20ba73c4db0757ef1772574dd593ec
- [x] check Material theming in M3 pickers
- [x] drop semantic role from Surface - https://github.com/androidx/androidx/commit/bf36f3091a4a1c71718c51dc55833ec4e0f30f3a
- ...

### PR DESCRIPTION
**Pull request by hrach**
closes #261

### CODE REVIEW

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/LinearIndeterminateProgressIndicator.kt [hunk 1]
```diff
@@ -39,100 +26,14 @@ public fun LinearIndeterminateProgressIndicator(
...
-        0f,
-        1f,
-        infiniteRepeatable(
-            animation = keyframes {
-                durationMillis = LinearAnimationDuration
-                0f at FirstLineHeadDelay with FirstLineHeadEasing
-                1f at FirstLineHeadDuration + FirstLineHeadDelay
-            },
-        ),
+    androidx.compose.material3.LinearProgressIndicator(
```
Comment (hrach) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/LinearIndeterminateProgressIndicator.kt:29]: Previously, the cap was counted to the size. Now it seems that the default impl doesn't take it into account. But less code on our side is IMO good. 
obrazek

**Review by Drekorian — COMMENTED** [review]
Looks good to me. 👍

**Review by Drekorian — APPROVED** [review]

File: catalog/build.gradle.kts [hunk 2]
```diff
@@ -121,6 +122,8 @@ kotlinter {
 }
 
 dependencies {
+    api(platform(libs.compose.bom))
```
Comment (shanio) [inline catalog/build.gradle.kts:125]: Catalog most probablyt does not need to expose the compose dependencies with api.

**Review by shanio — APPROVED** [review]

File: catalog/build.gradle.kts (hunk 2 above)
Comment (hrach) [reply catalog/build.gradle.kts]: Changed to implementation, though I'm not sure if it changes anything.

**Pull request merged**